### PR TITLE
dev:Anser Product Service is done,and modify API file errors.

### PR DESCRIPTION
**Pull request by kai890707**
**Description**

1. Anser product service is implement.
2.  Modify API file errors.

**Checklist:**
- [ ] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [x] Conforms to style guide

### CODE REVIEW

File: develop/app/Anser/Orchestrators/v2/CreateOrderOrchestrator.php [hunk 1]
```diff
@@ -29,7 +29,8 @@ protected function definition()
 
         // $this->setCacheInstance($this->cache);
         // $this->setCacheOrchestratorKey('createOrder_1');
-        $this->setStep()->addAction("product_service", $this->productService->getAllProduct());
+        $this->setStep()->addAction("product_service", $this->productService->deleteProduct(22));
```
Comment (ping-yee) [inline develop/app/Anser/Orchestrators/v2/CreateOrderOrchestrator.php:32]: Is this step really one of our actual orchestrator steps?
And we probably haven't started design the orchestrator yet.

File: develop/app/Anser/Services/v2/ProductService.php [hunk 2]
```diff
@@ -16,9 +16,64 @@ class ProductService extends SimpleService
...
-    public function getAllProduct()
+    /**
+     * Get all product
+     *
+     * @param integer|null $limit
+     * @param integer|null $offset
+     * @param string|null $isDesc
+     * @return ActionInterface
+     */
+    public function getAllProduct(?int $limit, ?int $offset, ?string $isDesc): ActionInterface
```
Comment (ping-yee) [inline develop/app/Anser/Services/v2/ProductService.php:27]: If there are the `null` type of parameters, I strongly recommend you should defind these parameters the initial values for them. shown below:
> public function getAllProduct(?int $limit = null, ?int $offset = null, ?string $isDesc = null): ActionInterface

Then the user can can this method without any parameters, like: `getAllProduct()`.

**Review by ping-yee — CHANGES_REQUESTED** [review]
Something need to be modified.

File: develop/app/Anser/Orchestrators/v2/CreateOrderOrchestrator.php (hunk 1 above)
Comment (kai890707) [reply develop/app/Anser/Orchestrators/v2/CreateOrderOrchestrator.php]: This is my test action to see if this action can be called correctly and this error has been fixed.

File: develop/app/Anser/Services/v2/ProductService.php (hunk 2 above)
Comment (kai890707) [reply develop/app/Anser/Services/v2/ProductService.php]: This change is done.

File: develop/app/Anser/Orchestrators/v2/CreateOrderOrchestrator.php [hunk 3]
```diff
@@ -30,6 +30,7 @@ protected function definition()
         // $this->setCacheInstance($this->cache);
         // $this->setCacheOrchestratorKey('createOrder_1');
         $this->setStep()->addAction("product_service", $this->productService->getAllProduct());
+
```
Comment (ping-yee) [inline develop/app/Anser/Orchestrators/v2/CreateOrderOrchestrator.php:33]: Remove this new line.

**Review by ping-yee — COMMENTED** [review]
Have a little change need to be handled.

**Review by ping-yee — APPROVED** [review]

**Pull request merged**